plugin_injector/plugin_injector.cc

```diff
@@ -147,6 +147,9 @@ class LLVMSemanticExtension : public SemanticExtension {
                     switch (send->fun.rawId()) {
                         case core::Names::keepForIde().rawId():
                         case core::Names::keepForTypechecking().rawId():
+                        // This method exists only to make inference work in certain
+                        // cases and doesn't actually compute anything.
+                        case core::Names::mustBeTruthy().rawId():
```

Comment:
ooc, why do this here instead of with a no-op intrinsic?

Reply:
Because a no-op intrinsic isn't really a no-op: it still assigns into the alias associated with the the class variable.  And if we want to get the right value to do that, we still have to get the value associated with the class variable.  Which is a whole bunch of unnecessary overhead.

I will admit this is not super clean, and it's also potentially bad: if we use the intrinsic in other contexts, it might not be safe to elide the intrinsic entirely.  There is probably a better way to do this, but I'm not sure what it is.  Maybe this is another reasons to have a real compiler-specific IR?

Reply:
> it still assigns into the alias associated with the the class variable

Note that this is a class variable set (`rb_cvar_set`, which is not super-cheap, since class variables don't have caches associated with them the way instance variables do).